Tests/Fixtures/Generated/CoreData/swift4-context-defaults.swift

```diff
@@ -70,11 +70,56 @@ internal class MainEntity: NSManagedObject {
   @NSManaged internal var boolean: Bool
   @NSManaged internal var date: Date?
   @NSManaged internal var decimal: NSDecimalNumber?
-  @NSManaged internal var double: Double
+  internal var double: Double? {
```

Comment:
If I understand the code correctly, the optional version is:
```swift
@NSManaged internal var double: Double?
```
This does not need the same custom getter/setter?

Maybe we should have `optional(Double|Int16|Int64)` attributes in the tests to show the corresponding generated code?

Reply:
Note that I changed the model so that some of the scalar attributes are optionals (see https://github.com/SwiftGen/SwiftGen/pull/567/files#diff-a8d5cc54c3ee6120761c601a66578206L18).

As I understand it, custom setters/getters are needed for scalar optionals as objc can't represent them.

Reply:
See http://aplus.rs/2017/handling-core-data-optional-scalar-attributes/